### ISSUE TITLE
acc 0.0

### ISSUE THREAD

**Issue by CaptainEven** [ronghuaiyang/arcface-pytorch#10]
=> total time is 42.86408877372742, average time is 0.1667863376409627
=> lfw face verification accuracy:  0.8643333333333333 threshold:  0.98421407
Tue Oct 16 14:23:41 2018 train epoch 7 iter 75 1.8792263763362134 iters/s loss 14.678580284118652 acc 0.0
Tue Oct 16 14:23:51 2018 train epoch 7 iter 175 10.736604388208807 iters/s loss 14.29747486114502 acc 0.0
Tue Oct 16 14:24:00 2018 train epoch 7 iter 275 10.673344835993447 iters/s loss 15.466602325439453 acc 0.0
Tue Oct 16 14:24:10 2018 train epoch 7 iter 375 10.266020200663386 iters/s loss 15.988080978393555 acc 0.0
Tue Oct 16 14:24:20 2018 train epoch 7 iter 475 10.013510335681348 iters/s loss 14.602261543273926 acc 0.0
Tue Oct 16 14:24:30 2018 train epoch 7 iter 575 9.911183891830836 iters/s loss 15.66580581665039 acc 0.0
Tue Oct 16 14:24:40 2018 train epoch 7 iter 675 10.237420150127207 iters/s loss 15.294169425964355 acc 0.0
Tue Oct 16 14:24:49 2018 train epoch 7 iter 775 10.42534408847289 iters/s loss 15.03502082824707 acc 0.0
Tue Oct 16 14:24:59 2018 train epoch 7 iter 875 10.592710894452473 iters/s loss 15.485675811767578 acc 0.0
Tue Oct 16 14:25:08 2018 train epoch 7 iter 975 10.55012234010108 iters/s loss 14.71179485321045 acc 0.0
Tue Oct 16 14:25:18 2018 train epoch 7 iter 1075 10.474356568004414 iters/s loss 14.181792259216309 acc 0.0
Tue Oct 16 14:25:28 2018 train epoch 7 iter 1175 9.514813052351975 iters/s loss 15.342684745788574 acc 0.0
Tue Oct 16 14:25:38 2018 train epoch 7 iter 1275 10.08869761467461 iters/s loss 14.384868621826172 acc 0.0
Tue Oct 16 14:25:48 2018 train epoch 7 iter 1375 10.343309102628595 iters/s loss 15.508356094360352 acc 0.0

**Comment by ronghuaiyang** [comment]
the acc calculation is wrong, you can ignore it.

**Comment by CaptainEven** [comment]
@ronghuaiyang Ok, thx for replying!

**Comment by CaptainEven** [comment]
Is train_list.txt formatted like this correct?
/mnt/diskc/even/WebFace/0124920/003.jpg	10572
/mnt/diskc/even/WebFace/0124920/012.jpg	10572
/mnt/diskc/even/WebFace/0124920/020.jpg	10572
/mnt/diskc/even/WebFace/2257466/045.jpg	10573
/mnt/diskc/even/WebFace/2257466/017.jpg	10573
/mnt/diskc/even/WebFace/2257466/011.jpg	10573
/mnt/diskc/even/WebFace/2257466/037.jpg	10573
/mnt/diskc/even/WebFace/2257466/055.jpg	10573
/mnt/diskc/even/WebFace/2257466/038.jpg	10573
/mnt/diskc/even/WebFace/2257466/054.jpg	10573
/mnt/diskc/even/WebFace/2257466/002.jpg	10573
/mnt/diskc/even/WebFace/2257466/053.jpg	10573

**Comment by ronghuaiyang** [comment]
yes

**Comment by CaptainEven** [comment]
Hi, lfw face verification accuracy:  0.919 after 60 epoches on webface, is something wrong?
Thu Oct 18 05:31:36 2018 train epoch 59 iter 23575 9.423660458263614 iters/s loss 13.846029281616211 acc 0.0
Thu Oct 18 05:31:47 2018 train epoch 59 iter 23675 8.841370517152965 iters/s loss 14.113338470458984 acc 0.0
Thu Oct 18 05:31:58 2018 train epoch 59 iter 23775 9.045688436231796 iters/s loss 14.169363021850586 acc 0.0
Thu Oct 18 05:32:09 2018 train epoch 59 iter 23875 9.078215778959866 iters/s loss 13.657951354980469 acc 0.0
Thu Oct 18 05:32:21 2018 train epoch 59 iter 23975 8.872302134609171 iters/s loss 14.437204360961914 acc 0.0
Thu Oct 18 05:32:31 2018 train epoch 59 iter 24075 9.336993715728159 iters/s loss 14.934171676635742 acc 0.0
Thu Oct 18 05:32:42 2018 train epoch 59 iter 24175 9.424477377691947 iters/s loss 13.678863525390625 acc 0.0
Thu Oct 18 05:32:52 2018 train epoch 59 iter 24275 9.611342413583303 iters/s loss 14.37238597869873 acc 0.0
Thu Oct 18 05:33:03 2018 train epoch 59 iter 24375 9.663595990142555 iters/s loss 14.07996654510498 acc 0.0
Thu Oct 18 05:33:13 2018 train epoch 59 iter 24475 9.445919208247018 iters/s loss 14.797931671142578 acc 0.0
Thu Oct 18 05:33:24 2018 train epoch 59 iter 24575 9.402641094293047 iters/s loss 13.223506927490234 acc 0.0
(7701, 1024)
=> total time is 52.16912531852722, average time is 0.20299270551956117
=> lfw face verification accuracy:  0.919 threshold:  0.9753611

**Comment by Vermeille** [comment]
> the acc calculation is wrong, you can ignore it.

Why? How can I fix this?

**Comment by ronghuaiyang** [comment]
@CaptainEven @Vermeille i made a mistake here, the acc is 0.0 is not the calculation problem, the calulation is : 
acc = np.mean((output == label).astype(int))
i thought it's the ' int ' type that cause the 0.0 problem.  but it's not . The calculation methed is right.
The acc is 0.0 because the model is not converged yet. The arc-face will converge after several epochs, so the acc is 0.0 before the model converges.

**Comment by abcde-bit** [comment]
> Hi, lfw face verification accuracy: 0.919 after 60 epoches on webface, is something wrong? Thu Oct 18 05:31:36 2018 train epoch 59 iter 23575 9.423660458263614 iters/s loss 13.846029281616211 acc 0.0 Thu Oct 18 05:31:47 2018 train epoch 59 iter 23675 8.841370517152965 iters/s loss 14.113338470458984 acc 0.0 Thu Oct 18 05:31:58 2018 train epoch 59 iter 23775 9.045688436231796 iters/s loss 14.169363021850586 acc 0.0 Thu Oct 18 05:32:09 2018 train epoch 59 iter 23875 9.078215778959866 iters/s loss 13.657951354980469 acc 0.0 Thu Oct 18 05:32:21 2018 train epoch 59 iter 23975 8.872302134609171 iters/s loss 14.437204360961914 acc 0.0 Thu Oct 18 05:32:31 2018 train epoch 59 iter 24075 9.336993715728159 iters/s loss 14.934171676635742 acc 0.0 Thu Oct 18 05:32:42 2018 train epoch 59 iter 24175 9.424477377691947 iters/s loss 13.678863525390625 acc 0.0 Thu Oct 18 05:32:52 2018 train epoch 59 iter 24275 9.611342413583303 iters/s loss 14.37238597869873 acc 0.0 Thu Oct 18 05:33:03 2018 train epoch 59 iter 24375 9.663595990142555 iters/s loss 14.07996654510498 acc 0.0 Thu Oct 18 05:33:13 2018 train epoch 59 iter 24475 9.445919208247018 iters/s loss 14.797931671142578 acc 0.0 Thu Oct 18 05:33:24 2018 train epoch 59 iter 24575 9.402641094293047 iters/s loss 13.223506927490234 acc 0.0 (7701, 1024) => total time is 52.16912531852722, average time is 0.20299270551956117 => lfw face verification accuracy: 0.919 threshold: 0.9753611

Hi brother. I also have the problem How did you solve the problem at last? Thanks a lot!
![image](https://user-images.githubusercontent.com/59106543/226823777-7199947f-a590-41e3-beca-ed6b78c6051a.png)
